js/main.js

```diff
@@ -162,34 +219,62 @@ function sortAlphabetical(countryArray) {
 function handleSearch(event) {
   unrenderAll();
 
-  for (let i = 0; i < countries.length; i++) {
-    if (countries[i].name.common.toLowerCase().includes(event.target.value.toLowerCase())) {
-      $countryDeck.append(renderCountry(countries[i]));
+  if (changeSearch === true) {
+    for (let i = 0; i < data.savedCountries.length; i++) {
+      if (data.savedCountries[i].name.common.toLowerCase().includes(event.target.value.toLowerCase())) {
+        $countryDeck.append(renderCountry(data.savedCountries[i]));
+      }
+    }
+  } else {
+    for (let i = 0; i < countries.length; i++) {
+      if (countries[i].name.common.toLowerCase().includes(event.target.value.toLowerCase())) {
+        $countryDeck.append(renderCountry(countries[i]));
+      }
     }
   }
 }
 
-// Function that unrenders every country
-function unrenderAll() {
-  const $countryWrappers = document.querySelectorAll('div.country-wrapper');
-
-  $countryWrappers.forEach(function (element) {
-    element.remove();
-  });
-}
-
 // Function that handles clicking events on cards
 function handleDeck(event) {
-  if (event.target.closest('.card') === null) {
+  const $countryClicked = event.target.closest('.card');
+  if ($countryClicked === null) {
     return;
   }
-  const $clickedCard = event.target.closest('.card');
-  $clickedCard.classList.toggle('is-flipped');
+  const $frontPlanePin = $countryClicked.querySelector('.country').querySelector('i');
+
+  if (event.target.matches('button')) {
+    event.target.classList.add('hidden');
+    $frontPlanePin.classList.remove('hidden');
+    data.savedCountries.push(getCountryFromCCA3($countryClicked.getAttribute('data-cca3')));
+  } else {
+    $countryClicked.classList.toggle('is-flipped');
+  }
+}
+
+// Formats population for front side render
+function formatPopulation(number) {
+  if (number > 1000000000) {
+    return (Math.round(number / 100000000) / 10) + ' Billion People';
+  } else if (number > 100000000) {
+    return (Math.round(number / 1000000)) + ' Million People';
+  } else if (number > 1000000) {
+    return (Math.round(number / 100000) / 10) + ' Million People';
+  } else {
+    return '< 1 Million People';
+  }
+}
+
+function getCountryFromCCA3(cca3) {
+  for (let i = 0; i < countries.length; i++) {
+    if (countries[i].cca3 === cca3) {
+      return countries[i];
+    }
+  }
 }
 
 // Event listeners
 $countryDeck.addEventListener('click', handleDeck);
-$search.addEventListener('input', handleSearch);
-
-// Run on startup
-getAllCountries();
+$searchBar.addEventListener('input', handleSearch);
+$switchToBucket.addEventListener('click', function () { viewSwap('bucketList'); });
+$switchToHome.addEventListener('click', function () { viewSwap('home'); });
+document.addEventListener('DOMContentLoaded', getAllCountries());
```

Comment:
This isn't actually passing `getAllCountries` as a callback function, it's going to call and execute this function immediately. You can either

1. Remove the `DOMContentLoaded` event and just call `getAllCountries()` directly
1. Do something similar as the lines above it, where you create an anonymous function, and call `getAllCountries` inside of it.